config/network-policies.yaml

```diff
@@ -0,0 +1,566 @@
+          app.kubernetes.io/name: cf-api-clock
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-worker
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: capi-kpack-watcher
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Egress
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: cf-api-kpack-watcher
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: routecontroller
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Egress
+  podSelector:
+    matchLabels:
+      app: routecontroller
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: log-cache
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      app: log-cache
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          istio: ingressgateway
+      namespaceSelector:
+        matchLabels:
+          cf-for-k8s.cloudfoundry.org/istio-system-ns: ""
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app: fluentd
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+  egress:
+  - to:
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: uaa
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: metric-proxy
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Ingress
+  - Egress
+  podSelector:
+    matchLabels:
+      app: metric-proxy
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          app.kubernetes.io/name: cf-api-server
+      namespaceSelector:
+        matchLabels:
+          name: #@ data.values.system_namespace
+  egress:
+  - {}
+---
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: fluentd
+  namespace: #@ data.values.system_namespace
+spec:
+  policyTypes:
+  - Egress
+  podSelector:
+    matchLabels:
+      app: fluentd
+  egress:
+  - {}
```

Comment:
This is wide-open. Originally I had `fluentd` only allowed to egress to `log-cache`, but it was trying to make an `http` call to something. Would appreciate help on this one.

Reply:
@tcdowney at some point(in the not so far future) we plan on making fluentd able to egress to anywhere specified off platform, but I think only to log-cache should be fine at the moment

Reply:
Let me rephrase, I was remembering wrong. So, for fluentd. Fluentd needs to contact(at the moment), the kubernetes api, and also log-cache(fluentd initiates), and will in the future need to contact any user defined source(so contact all makes sense. If there was a way to unlimit external ip's, or for us to dynamically limit ips(might be very difficult and not worth it), maybe we could, but the way I see it unlimited egress from fluentd makes sense to me, then we could hypotheticially limit internal sources.

as far as ingress though, we'll need teams to be able to ingress into fluentd. Obviously unlimited would work, but if we want to limit it further that could work too?

Reply:
Is this still an open question? If not, could someone please update this thread and resolve it?

Reply:
Summarizing what was said:
Since `fluentd` calls out to the Kubernetes API it needs open egress here since we don't know if the API is hosted on cluster or external to the cluster. Additionally it sounds like `fluentd` Open egress here is correct, at least for now.

Since components are sending stuff directly to `fluentd` it sounds like we're missing some allow rules for `ingress` here.

I don't think we want `ingress` to be truly unlimited from all namespaces (I think it would be bad if apps can arbitrarily send stuff to log-cache... might be able to pollute the logs for other apps), so I'll add rules for the `cf-api-*` jobs and `istio-ingressgateway`.. I think those are the things that are or will soon be sending logs directly to `fluentd` via the network.

Reply:
Added these `ingress` rules:
https://github.com/cloudfoundry/cf-for-k8s/pull/270/commits/b9b42d035ae86440f5648bfaf4bdfb5355cde6e9